core/src/com/unciv/ui/trade/ExpanderTab.kt

```diff
@@ -1,45 +1,100 @@
+        const val fontSize = 24
+        const val arrowSize = 18f
+        const val arrowImage = "OtherIcons/BackArrow"
+        val arrowColor = Color(1f,0.96f,0.75f,1f)
+        const val animationDuration = 0.2f
+    }
+
+    private val header = Table(skin)  // Header with label and icon, touchable to show/hide
+    private val headerLabel = title.toLabel(fontSize = fontSize)
+    private val headerIcon = ImageGetter.getImage(arrowImage)
+    private val contentWrapper = Table()  // Wrapper for innerTable, this is what will be shown/hidden
+
+    /** The container where the client should add the content to toggle */
+    val innerTable = Table()
+
+    /** Indicates whether the contents are currently shown */
+    var isOpen = initialOpen
+        private set(value) {
+            if (value == field) return
+            field = value
+            update()
         }
-        add(toggle).fill().row()
-        tab.add(innerTable).pad(10f)
-        add(tab)
+
+    init {
+        header.defaults().pad(10f)
+        headerIcon.setSize(arrowSize, arrowSize)
+        headerIcon.setOrigin(Align.center)
+        headerIcon.rotation = 180f
+        headerIcon.color = arrowColor
+        header.background(ImageGetter.getBackground(ImageGetter.getBlue()))
+        header.add(headerLabel)
+        header.add(headerIcon).size(arrowSize).align(Align.center)
+        header.touchable= Touchable.enabled
+        header.onClick { toggle() }
+        add(header).expandX().fill().row()
+        contentWrapper.defaults().pad(defaultPad)
+        add(contentWrapper).expandX()
+        initContent?.invoke(innerTable)
+        update(noAnimation = true)
     }
 
-    fun close(){
-        if(!isOpen) return
-        toggle.clearChildren()
-        toggle.add("- $title".toLabel(fontSize = 24))
-        tab.clear()
-        isOpen=false
+    private fun update(noAnimation: Boolean = false) {
+        if (noAnimation || !UncivGame.Current.settings.continuousRendering) {
+            contentWrapper.clear()
+            if (isOpen) contentWrapper.add(innerTable)
+            headerIcon.rotation = if (isOpen) 90f else 180f
+            return
+        }
+        val action = object: FloatAction ( 90f, 180f, animationDuration, Interpolation.linear) {
+            override fun update(percent: Float) {
+                super.update(percent)
+                headerIcon.rotation = this.value
+                if (this.isComplete) {
+                    contentWrapper.clear()
+                    if (isOpen) contentWrapper.add(innerTable)
+                }
+            }
+        }.apply { isReverse = isOpen }
+        addAction(action)
     }
 
-    fun open(){
-        if(isOpen) return
-        toggle.clearChildren()
-        toggle.add("+ $title".toLabel(fontSize = 24))
-        tab.add(innerTable)
-        isOpen=true
+    fun close() {
+        isOpen = false
+    }
+    fun open() {
+        isOpen = true
+    }
+    fun toggle() {
+        isOpen = !isOpen
     }
```

Comment:
Don't pre-create unused functions. If we need them, we can make them later.

Reply:
Happily - but two were already there before I started and `toggle` is used. Will patch later.